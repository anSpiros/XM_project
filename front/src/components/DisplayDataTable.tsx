import React from "react";
import { DisplayDataInterface } from "../interfaces/display-data.interface";

interface headerDataInterface {
  column: string;
}

interface DisplayDataTableProps {
  headerData: headerDataInterface[];
  bodyData: DisplayDataInterface[];
}

const shouldRender = (
  headerData: headerDataInterface[],
  bodyData: DisplayDataInterface[]
) => {
  return headerData.length > 0 && bodyData.length > 0;
};

function DisplayDataTable({ headerData, bodyData }: DisplayDataTableProps) {
  return (
    <>
      {shouldRender(headerData, bodyData) ? (
        <div className="table-responsive">
          <table className="table">
            <thead>
              <tr>
                {headerData.map((data, index: number) => (
                  <th key={index}>{data.column}</th>
                ))}
              </tr>
            </thead>
            <tbody>
              {bodyData.map((data, index: number) => {
                return (
                  <tr key={index}>
                    <td>{data.date}</td>
                    <td>{data.open}</td>
                    <td>{data.high}</td>
                    <td>{data.low}</td>
                    <td>{data.close}</td>
                    <td>{data.volume}</td>
                  </tr>
                );
              })}
            </tbody>
          </table>
        </div>
      ) : null}
    </>
  );
}

export default DisplayDataTable;
