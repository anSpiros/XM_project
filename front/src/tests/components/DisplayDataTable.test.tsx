import React from "react";
import { render } from "@testing-library/react";
import DisplayDataTable from "../../components/DisplayDataTable";
import {
  DisplayDataInterface,
  headerData,
} from "../../interfaces/display-data.interface";

describe("DisplayData component", () => {
  test("should render correctly without data", () => {
    const { asFragment } = render(
      <DisplayDataTable headerData={[]} bodyData={[]} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("should render correctly with data", () => {
    const header: headerData[] = [
      {
        column: "Date",
      },
      {
        column: "Open",
      },
      {
        column: "High",
      },
      {
        column: "Low",
      },
      {
        column: "Close",
      },
      {
        column: "Volume",
      },
    ];
    const body: DisplayDataInterface[] = [
      {
        date: "test",
        open: 2,
        high: 2,
        low: 2,
        close: 1,
        volume: 1,
      },
    ];
    const { asFragment } = render(
      <DisplayDataTable headerData={header} bodyData={body} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
