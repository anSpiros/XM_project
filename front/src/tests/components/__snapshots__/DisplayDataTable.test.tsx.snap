// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayData component should render correctly with data 1`] = `
<DocumentFragment>
  <div
    class="table-responsive"
  >
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            Date
          </th>
          <th>
            Open
          </th>
          <th>
            High
          </th>
          <th>
            Low
          </th>
          <th>
            Close
          </th>
          <th>
            Volume
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            test
          </td>
          <td>
            2
          </td>
          <td>
            2
          </td>
          <td>
            2
          </td>
          <td>
            1
          </td>
          <td>
            1
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`DisplayData component should render correctly without data 1`] = `<DocumentFragment />`;
